ksql-functional-tests/src/test/resources/query-validation-tests/delimited-value-format.json

```
@@ -0,0 +1,222 @@
+        }
+      ],
+      "expectedException": {
+        "type": "io.confluent.ksql.util.KsqlException",
+        "message": "Error in WITH clause property 'VALUE_DELIMITER': Delimiter must be a single character, 'TAB' or 'SPACE'.\nExample valid value: ';'"
+      },
+      "inputs": [],
+      "outputs": []
+    },
+    {
+      "name": "validate delimeter is not empty",
+      "statements": [
+        "CREATE STREAM TEST WITH (kafka_topic='test_topic', value_format='DELIMITED', value_delimiter='');"
+      ],
+      "topics": [
+        {
+          "name": "test_topic",
+          "format": "DELIMITED"
+        }
+      ],
+      "expectedException": {
+        "type": "io.confluent.ksql.util.KsqlException",
+        "message": "Error in WITH clause property 'VALUE_DELIMITER': Delimiter cannot be empty or whitespace."
+      },
+      "inputs": [],
+      "outputs": []
+    },
+    {
+      "name": "validate delimeter is not a space",
+      "statements": [
+        "CREATE STREAM TEST WITH (kafka_topic='test_topic', value_format='DELIMITED', value_delimiter=' ');"
+      ],
+      "topics": [
+        {
+          "name": "test_topic",
+          "format": "DELIMITED"
+        }
+      ],
+      "expectedException": {
+        "type": "io.confluent.ksql.util.KsqlException",
+        "message": "Error in WITH clause property 'VALUE_DELIMITER': Delimiter cannot be empty or whitespace."
+      },
+      "inputs": [],
+      "outputs": []
+    },
+    {
+      "name": "validate delimeter is not a tab character",
+      "statements": [
+        "CREATE STREAM TEST WITH (kafka_topic='test_topic', value_format='DELIMITED', value_delimiter='\t');"
+      ],
+      "topics": [
+        {
+          "name": "test_topic",
+          "format": "DELIMITED"
+        }
+      ],
+      "expectedException": {
+        "type": "io.confluent.ksql.util.KsqlException",
+        "message": "Error in WITH clause property 'VALUE_DELIMITER': Delimiter cannot be empty or whitespace."
+      },
+      "inputs": [],
+      "outputs": []
+    },
+    {
+      "name": "select delimited value_format with pipe separated values using custom delimiter character",
+      "statements": [
+        "CREATE STREAM TEST (ID bigint, NAME varchar, VALUE integer) WITH (kafka_topic='test_topic', value_format='DELIMITED', value_delimiter='|');",
+        "CREATE STREAM S2 as SELECT id, name, value FROM test;"
+      ],
+      "topics": [
+        {
+          "name": "test_topic",
+          "format": "DELIMITED"
+        }
+      ],
+      "inputs": [
+        {"topic": "test_topic", "key": 0, "value": "0|zero|0", "timestamp": 0},
+        {"topic": "test_topic", "key": 0, "value": "0|100|100", "timestamp": 0},
+        {"topic": "test_topic", "key": 100, "value": "100|100|500", "timestamp": 0},
+        {"topic": "test_topic", "key": 100, "value": "100|100|100", "timestamp": 0}
+      ],
+      "outputs": [
+        {"topic": "S2", "key": 0, "value": "0,zero,0", "timestamp": 0},
```

I would say that the output should use the same delimiter as the input unless explicitly overridden, so this should be 

```suggestion
        {"topic": "S2", "key": 0, "value": "0|zero|0", "timestamp": 0},
```

This would then follow the same pattern as other WITH cause items, e.g. VALUE_FORMAT.